packages/apollo-graphql/src/transforms.ts

```diff
@@ -94,13 +95,37 @@ function sorted<T>(
   return undefined;
 }
 
+// Predicate for filtering and type checking operation definitions (query, mutation, subscription)
+// Similar to graphql's isExecutableDefinitionNode
+function isOperationDefinitionNode(
+  node: ASTNode,
+): node is OperationDefinitionNode {
+  return node.kind === 'OperationDefinition';
+}
+
+// Predicate for filtering and type checking fragment definitions
+// Similar to graphql's isExecutableDefinitionNode
+function isFragmentDefinitionNode(
+  node: ASTNode,
+): node is FragmentDefinitionNode {
+  return node.kind === 'FragmentDefinition';
+}
+
 // sortAST sorts most multi-child nodes alphabetically. Using this as part of
 // your signature calculation function may make it easier to tell the difference
 // between queries that are similar to each other, and if for some reason your
 // GraphQL client generates query strings with elements in nondeterministic
 // order, it can make sure the queries are treated as identical.
 export function sortAST(ast: DocumentNode): DocumentNode {
-  return visit(ast, {
+  const [operation] = ast.definitions.filter(isOperationDefinitionNode);
+  const fragments = ast.definitions.filter(isFragmentDefinitionNode);
+
+  const astWithSortedDefinitions = {
+    ...ast,
+    definitions: [operation, ...sortBy(fragments, 'name.value')],
+  };
+
+  return visit(astWithSortedDefinitions, {
```

Comment:
This is excellent. See: d341e08691879704c429dea42da0d6f9220efec3

I'd like your take on the iteratees that I've added.
* `operation` will give a nice grouping of `query` `mutation` `subscription`. I don't see this adding any value to the normalization itself, but I imagine it will read much more nicely. Do we expect human eyes on these at any point? I don't know the answer to that.
* The stringification for anonymous operations - after reading your explanation above about single anonymous operations only, considering this edge case may be overkill.